cmd/ooniprobe/internal/nettests/dnscheck.go

```diff
@@ -1,60 +1,42 @@
 package nettests
 
 import (
-	"encoding/json"
+	"context"
 
-	"github.com/ooni/probe-cli/v3/internal/engine/experiment/dnscheck"
-	"github.com/ooni/probe-cli/v3/internal/engine/experiment/run"
-	"github.com/ooni/probe-cli/v3/internal/runtimex"
+	engine "github.com/ooni/probe-cli/v3/internal/engine"
+	"github.com/ooni/probe-cli/v3/internal/engine/model"
 )
 
 // DNSCheck nettest implementation.
 type DNSCheck struct{}
 
-var dnsCheckDefaultInput []string
-
-func dnsCheckMustMakeInput(input *run.StructuredInput) string {
-	data, err := json.Marshal(input)
-	runtimex.PanicOnError(err, "json.Marshal failed")
-	return string(data)
-}
-
-func init() {
-	// The following code just adds a minimal set of URLs to
-	// test using DNSCheck, so we start exposing it.
-	//
-	// TODO(bassosimone):
-	//
-	// 1. we should be getting input from the backend instead of
-	// having an hardcoded list of inputs here.
-	//
-	// 2. we should modify dnscheck to accept http3://... as a
-	// shortcut for https://... with h3. If we don't do that, we
-	// are stuck with the h3 results hiding h2 results in OONI
-	// Explorer because they use the same URL.
-	//
-	// 3. it seems we have the problem that dnscheck results
-	// appear as the `run` nettest in `ooniprobe list <ID>` because
-	// dnscheck is run using the `run` functionality.
-	dnsCheckDefaultInput = append(dnsCheckDefaultInput, dnsCheckMustMakeInput(
-		&run.StructuredInput{
-			DNSCheck: dnscheck.Config{},
-			Name:     "dnscheck",
-			Input:    "https://dns.google/dns-query",
-		}))
-	dnsCheckDefaultInput = append(dnsCheckDefaultInput, dnsCheckMustMakeInput(
-		&run.StructuredInput{
-			DNSCheck: dnscheck.Config{},
-			Name:     "dnscheck",
-			Input:    "https://cloudflare-dns.com/dns-query",
-		}))
+func (n DNSCheck) lookupURLs(ctl *Controller) ([]string, error) {
+	inputloader := &engine.InputLoader{
+		CheckInConfig: &model.CheckInConfig{
+			// not needed because we have default input
+		},
+		ExperimentName: "dnscheck",
+		InputPolicy:    engine.InputOrStaticDefault,
+		Session:        ctl.Session,
+		SourceFiles:    ctl.InputFiles,
+		StaticInputs:   ctl.Inputs,
+	}
+	testlist, err := inputloader.Load(context.Background())
+	if err != nil {
+		return nil, err
+	}
+	return ctl.BuildAndSetInputIdxMap(ctl.Probe.DB(), testlist)
```

Comment:
Explanation of the changes in this file: `DNSCheck` now has a `lookupURLs` method that uses the engine's `InputLoader` with the `InputOrStaticDefault` policy. The end result is that we use `--input` or `--input-files` if they are provided. Otherwise, we fallback to a static input list defined in the engine.